### PR TITLE
Panel Menu: Use config explore enabled as an override to access control 

### LINKED ISSUE #56989: Explore button on panel is visible if explore function is disabled
**What happened**:

If the `explore` feature is disabled the `Explore` button is still visible on the panel.

image

Furthermore the keyboard shortcut `x` still works and redirects to:

image

**What you expected to happen**:

The `Explore` button should be hidden like in Grafana 8.x

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a new dashboard
2. Add a panel
3. Explore button is available in menu

Docker-Compose file:

```yaml
version: '3'

services:
  grafana:
    image: grafana/grafana:9.2.0
    environment:
      GF_SECURITY_ADMIN_USER: admin
      GF_SECURITY_ADMIN_PASSWORD: test
      GF_EXPLORE_ENABLED: false
    ports:
      - 3000:3000
```

**Environment**:
- Grafana version: `9.2.0`

### PR DESCRIPTION
**Pull request by gelicia**
**What this PR does / why we need it**:
I'm not actually sure this should be correct. But generally if someone has explicitly turned off explore in their config, they will probably intend for that setting to override access control. That said, this might be a discussion where a decision was already made.

If the existing logic is working as designed, please clarify that in the issue ticket. Thank you! 

**Which issue(s) this PR fixes**:
Fixes #56989

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gamab — APPROVED** [review]
LGTM. This is inline with the backend check on the NavLink: https://github.com/grafana/grafana/blob/8949e574e6288671343b80991479e2e2b9adfc88/pkg/services/navtree/navtreeimpl/navtree.go#L117-L119

@kalleep maybe we can even remove the `ActionDatasourcesExplore` grant based on the config option?
https://github.com/grafana/grafana/blob/8949e574e6288671343b80991479e2e2b9adfc88/pkg/api/accesscontrol.go#L56-L73

**Comment by gelicia** [comment]
Sorry @gamab - I thought it might be preferable to do the check on if the flag was explicitly set to false - can you check it out again real quick? I can also put it back if you preferred a simpler check.

File: public/app/core/services/context_srv.ts [hunk 1]
```diff
@@ -157,9 +157,9 @@ export class ContextSrv {
 
   hasAccessToExplore() {
     if (this.accessControlEnabled()) {
-      return this.hasPermission(AccessControlAction.DataSourcesExplore);
+      return this.hasPermission(AccessControlAction.DataSourcesExplore) && config.exploreEnabled;
     }
-    return (this.isEditor || config.viewersCanEdit) && config.exploreEnabled;
+    return (this.isEditor || config.viewersCanEdit) && !(config.exploreEnabled === false);
```
Comment (gamab) [inline public/app/core/services/context_srv.ts:162]:
```suggestion
    return (this.isEditor || config.viewersCanEdit) && (config.exploreEnabled !== false);
```

**Review by gamab — COMMENTED** [review]
Just a question what is the reason with the explicit check on line 162 and non-explicit check on 160?

File: public/app/core/services/context_srv.ts (hunk 1 above)
Comment (gelicia) [reply public/app/core/services/context_srv.ts]: Youre absolutely right, we should just stick with the existing simple logic check. I changed it back to what it was.

**Pull request merged**